includes/class-wc-cache-helper.php

```diff
@@ -42,14 +42,34 @@ public static function init() {
 	 */
 	public static function additional_nocache_headers( $headers ) {
 		$agent = isset( $_SERVER['HTTP_USER_AGENT'] ) ? wp_unslash( $_SERVER['HTTP_USER_AGENT'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+
+		$set_cache = false;
+
 		/**
 		 * Allow plugins to enable nocache headers. Enabled for Google weblight.
 		 *
-		 * @see   https://support.google.com/webmasters/answer/1061943?hl=en
 		 * @param bool $enable_nocache_headers Flag indicating whether to add nocache headers. Default: false.
 		 */
-		if ( false !== strpos( $agent, 'googleweblight' ) || apply_filters( 'woocommerce_enable_nocache_headers', false ) ) {
+		if ( apply_filters( 'woocommerce_enable_nocache_headers', false ) ) {
+			$set_cache = true;
+		}
+
+		/**
+		 * Enabled for Google weblight.
+		 *
+		 * @see https://support.google.com/webmasters/answer/1061943?hl=en
+		 */
+		if ( false !== strpos( $agent, 'googleweblight' ) ) {
 			// no-transform: Opt-out of Google weblight. https://support.google.com/webmasters/answer/6211428?hl=en.
+			$set_cache = true;
+		}
+
+		// Note that is_cart() will return true even for checkout page.
+		if ( false !== strpos( $agent, 'Chrome' ) && is_cart() ) {
```

Comment:
So just to be clear, do we want this to take effect on the checkout page?

* For me, `is_cart()` is only triggered on the cart page and this condition is not met on the checkout page. I'm not sure why we see different things there.
* If we want this to take effect on checkout pages, should we be explicit about it and use the `is_checkout()` function?

Reply:
This is because we already have a bug in the logic somewhere. If you look at this https://github.com/woocommerce/woocommerce/blob/5.3.0/includes/wc-conditional-functions.php#L98 you will see it also checks for `WOOCOMMERCE_CART` constant and if that is true, `is_cart()` will return true. Coincidentally that constant is set in checkout page as well. Hence this issue. But yeah we could explicitly also add `is_checkout()` as well.

Reply:
Yep, I think that makes sense—again for me, locally, `is_cart()` does not return true on the checkout page. So it would seem like that's a bug, as you say, and possibly a bug that needs specific conditions to be triggered (so either way, shouldn't be relied on)?

Reply:
Yeah I am not too sure myself. It might of been something wonky with cache but the way I found this is when I ran xdebug and stepped through. I've since removed that comment. So another round please @barryhughes